File: day1/day1.py
# Part 1
currentCal, maxCal, cal2, cal3, = 0, 0, 0, 0
file = open("day1.txt", 'r')

for line in file:
  if line == '\n':
    if currentCal > maxCal:
      maxCal = currentCal
    elif (currentCal < maxCal) & (currentCal > cal2):
      cal2 = currentCal
    elif (currentCal < cal2) & (currentCal > cal3):
      cal3 = currentCal
    currentCal = 0
    continue
  currentCal += int(line.strip())

print(maxCal)
maxCal += cal2+cal3
print(maxCal)

File: day1/day1.rs
use std::{
  fs::File,
  io::{BufReader, BufRead},
};

fn main() {
  let file: File = File::open("day1.txt").unwrap();
  let reader: BufReader<File> = BufReader::new(file);
  let lines = reader.lines();
  let mut ans: Vec<i32> = vec![];
  let mut temp_int: i32 = 0;

  for int in lines {
    let temp_string: String = int.unwrap();
    if temp_string == "" {
      ans.push(temp_int);
      temp_int = 0;
      continue;
    }
    temp_int += temp_string.parse::<i32>().unwrap();
  }

  ans.sort();
  ans.reverse();
  println!("Answer 1: {}", ans[0]);
  println!("Answer 2: {}", ans[0]+ans[1]+ans[2]);
  
}